insights/parsers/nginx_conf.py

```diff
@@ -0,0 +1,338 @@
+    index.htm index.php'], ['log_format', 'main  \'$remote_addr - $remote_user [$time_local] "$request" \'\n\'$status
+    $body_bytes_sent "$http_referer" \'\n\'"$http_user_agent" "$http_x_forwarded_for"\''], ['default_type',
+    'application/octet-stream'], ['access_log', 'logs/access.log  main'], ['sendfile', 'on'], ['tcp_nopush', 'on'],
+    ['server_names_hash_bucket_size', '128'], [['server'], [['listen', '80'], ['server_name', 'domain1.com
+    www.domain1.com'], ['access_log', 'logs/domain1.access.log  main'], ['root', 'html'], [['location', '~',
+    '\\.php$'], [['fastcgi_pass', '127.0.0.1:1025']]]]], [['server'], [['listen', '80'], ['server_name', 'domain2.com
+    www.domain2.com'], ['access_log', 'logs/domain2.access.log  main'], [['location', '~', '^/(images|javascript|
+    js|css|flash|media|static)/'], [['root', '/var/www/virtual/big.server.com/htdocs'], ['expires', '30d']]],
+    [['location', '/'], [['proxy_pass', 'http://127.0.0.1:8080']]]]], [['map', '$http_upgrade',
+    '$connection_upgrade'], [['default', 'upgrade'], ["''", 'close']]], [['upstream', 'websocket'], [['server',
+    '10.66.208.205:8010']]], [['upstream', 'big_server_com'], [['server', '127.0.0.3:8000 weight=5'], ['server',
+    '127.0.0.3:8001 weight=5'], ['server', '192.168.0.1:8000'], ['server', '192.168.0.1:8001']]], [['server'],
+    [['listen', '80'], ['server_name', 'big.server.com'], ['access_log', 'logs/big.server.access.log main'],
+    [['location', '/'], [['proxy_pass', 'http://big_server_com']]]]]]]]
+
+    After converting, it should be like following:
+
+    For the section, the section name would be a dict key, and content embraced by brace would be value. In some
+    sections, the first item is like ['location', '/'], in this case, the convert rule is that 'location' would be
+    the key, and add {"name":'/'} key-value to the dict value.
+    """
+    dict_all = {}
+    for item in li:
+        if depth(item) == 1:
+            dict_all[item[0]] = item[1]
+
+        if depth(item) == 3:
+            subdict_all = {}
+            for sub_item in item[1]:
+                subdict_all[sub_item[0]] = sub_item[1]
+            dict_all[item[0][0]] = subdict_all
+
+        if depth(item) == 5:
+            subdict_all = {}
+            for sub_item in item[1]:
+                if depth(sub_item) == 1:
+                    subdict_all[sub_item[0]] = sub_item[1]
+                if depth(sub_item) == 3:
+                    sub2dict_all = {}
+                    for sub2_item in sub_item[1]:
+                        sub2dict_all[sub2_item[0]] = sub2_item[1]
+                    if len(sub_item[0]) > 1:
+                        sub2dict_all["name"] = ' '.join(sub_item[0][1:])
+                    subdict_all[sub_item[0][0]] = _handle_key_value(subdict_all, sub_item[0][0], sub2dict_all)
+            dict_all[item[0][0]] = subdict_all
+
+        if depth(item) == 7:
```

Comment:
There are some duplicated code slices in this function, how about scratching the duplicated code slices and defining several private-inner-functions to reuse them?